src/test/ui/rfc-2565-param-attrs/param-attrs-pass.rs

```diff
@@ -0,0 +1,21 @@
+// compile-pass
+
+#![feature(param_attrs)]
+
+extern "C" {
+    fn ffi(#[cfg(bar)] a: u8, #[cfg(bar)] ...);
+}
+
+trait Trait2015 { fn bar(#[cfg(bar)] u8, #[cfg(bar)] u8); }
+trait Trait2018 { fn bar(#[cfg(bar)] a: u8, #[cfg(bar)] b: u8); }
+
+fn fn_ptr(#[cfg(bar)] a: u8, #[cfg(bar)] b: u8) { }
+
+struct Struct {}
+impl Struct {
+    fn struct_fn(#[cfg(bar)] self, #[cfg(bar)] a: u8) { }
```

Comment:
Please extend to more forms of receivers including:

```rust
&self
&mut self
&'a self
&'a mut self
self: Self
self: Box<Self>
```

also extend `struct_fn` to a trait implementation, but also in a trait definition.